### PR TITLE
Moved dotted named behaviors to named behaviors.

### PR DESCRIPTION
**Pull request by iham**

### CODE REVIEW

**Comment by iham** [comment]
@jenkins-plone-org please run jobs

**Comment by iham** [comment]
@jenkins-plone-org please run jobs

**Comment by iham** [comment]
@jensens, @mauritsvanrees or anybody else who might merge this before it is outdated again.
the errors thrown in the pull-request-5.2 are completely non-related to the changes/addons i made. merge it anyway?

File: plone/app/upgrade/v52/final.py [hunk 1]
```diff
@@ -27,3 +27,33 @@ def rebuild_redirections(context):
...
+
+    ptt = api.portal.get_tool('portal_types')
+
+    ftis = [fti for fti in ptt.objectValues() if IDexterityFTI.providedBy(fti)]
+
+    for fti in ftis:
+        behaviors = []
+        for behavior in fti.behaviors:
+            behavior_registration = lookup_behavior_registration(behavior)
+            behaviors.append(behavior_registration.name)
```
Comment (jensens) [inline plone/app/upgrade/v52/final.py]: Off-head: I think it is still possible to register behaviors w/o a name. In this case None would be appended.

Comment (iham) [reply plone/app/upgrade/v52/final.py]: good point - i did not take that into consideration. i will make that less strict

**Comment by iham** [comment]
@jenkins-plone-org please run jobs

**Comment by iham** [comment]
@jenkins-plone-org please run jobs

**Comment by iham** [comment]
@jensens i fixed the name thingy. still some archetypes related tests are failing, but i guess, those are not related to my code.

**Comment by jensens** [comment]
The AT-related failures are unelated. LGTM.

**Comment by jensens** [comment]
I now run both together, if green I would merge it:
```
https://github.com/plone/plone.app.contenttypes/pull/520
https://github.com/plone/plone.app.upgrade/pull/205
```

**Comment by mauritsvanrees** [comment]
The AT failures are indeed unrelated, and I have meanwhile fixed them by checking out Products.Archetypes. The cause of the failures went unnoticed because Jenkins had general problems earlier this week. So the jobs for these PR should be green when they finish.

For my understanding: why are named behaviors better then dotted behaviors? I am sure that you are right, but is this maybe in the docs?

And minor point: when I run the upgrade step twice, I get lots of lines like this:

```
Moved "plone.namefromtitle" to "plone.namefromtitle"
```

So the code is missing a condition like this:

```
if named_behavior == behavior:
    continue
```

**Comment by iham** [comment]
@mauritsvanrees your partially right.

i create a new list of behaviors based on the existing behaviors and replace fti.behaviors with a completely new tuple based on that list.
by simply continuing the loop i would not add it to the list and so it would be dropped - not good ;)

but:
i've changed the message output so you can see the correct action as this might be more help.

**Comment by iham** [comment]
@jenkins-plone-org please run jobs

**Comment by mauritsvanrees** [comment]
@iham, Ah you are right.
I mostly meant that we shouldn't print anything when nothing changes. But the extra info is fine here with your latest change.

**Comment by jensens** [comment]
> For my understanding: why are named behaviors better then dotted behaviors? I am sure that you are right, but is this maybe in the docs?

There are 3 main advantages:
1. easier to remember
2. no dotted-path lock-in, makes refactoring is simpler
3. can be overridden

**Pull request merged**